db/db_test2.cc

```diff
@@ -4482,8 +4526,9 @@ TEST_F(DBTest2, TestCompactFiles) {
   GetSstFiles(env_, dbname_, &files);
   ASSERT_EQ(files.size(), 2);
 
-  port::Thread user_thread1(
-      [&]() { db_->CompactFiles(CompactionOptions(), handle, files, 1); });
+  port::Thread user_thread1([&]() {
+    ASSERT_OK(db_->CompactFiles(CompactionOptions(), handle, files, 1));
```

Comment:
Another potential threading assert issue...

Reply:
I have made some changes now which are inline with the advice in https://github.com/facebook/rocksdb/pull/7754